test/libscap/CMakeLists.txt

```diff
@@ -44,9 +44,24 @@ set(LIBSCAP_TESTS_LIBRARIES "${GTEST_LIB}" "${GTEST_MAIN_LIB}" "${CMAKE_THREAD_L
 set(LIBSCAP_TESTS_DEPENDENCIES gtest scap)
 
 # Test suite asserting against pure userspace components
-file(GLOB_RECURSE USERSPACE_TEST_SUITE "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.cpp")
+file(GLOB_RECURSE USERSPACE_TEST_SUITE "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.cpp"
+	 "${CMAKE_CURRENT_SOURCE_DIR}/test_suites/userspace/*.c"
+)
 set(LIBSCAP_TESTS_SOURCES ${USERSPACE_TEST_SUITE})
 
+# Add _GNU_SOURCE definition for makedev and set C flags
+add_definitions(-D_GNU_SOURCE)
```

Comment:
This would add the definition globally, right? What about using https://cmake.org/cmake/help/latest/command/target_compile_definitions.html ?

Reply:
Can we remove this line then?